### PR TITLE
Complete tutor-tutorleaderboard-dev

### PR DESCRIPTION
**Pull request by johncoxdev**
### Description
---
We created tutoring command called `/tutor-leaderboard` in order for our tutors to see who has the top hours.

### Caveats
---
#### tutor-leaderboard.ts
`tutorleaderboardBody()` is a function that returns an embed back to the tutor. Users are allowed to see a range of top tutor hours from 1-20 (Default: 10). 
#### supabase.ts
`getTutorHours` is a function that returns the sum of hours a tutor has done. A new _"Remote Procedure Call"_ which stores the actual SQL query and then returns the number. `getTutorLeaderboard()` is a function that finds all of the tutor hours and stores them within a list. We fetch all of the hours from the `getTutorHours()`, we then start pushing it towards an array of dictionary(id, hours). We sort the array, and finally start creating the string for the embed. In order to get the active students, we have to check that within the logs that they are active tutors so we use the `getTutor()` function, and will continue if they're not an active tutor. After that we are then going to start putting the user in order within the embed and list them accordingly. 
### Pictures
---
Here is an example for options:
image
Then the output of the command would look like:
image

The full command with multiple people would look like this (with default value of 10):
image
image

### CODE REVIEW

File: src/commands/tutor-commands/tutor-leaderboard.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        .setName("number")
+        .setDescription("Number of leaderboard spaces you want to see!")
+        .setMaxValue(20)
+        .setMinValue(1)
+        .setRequired(false)
+    ),
+  run: async (interaction) => {
+    await interaction.deferReply({ ephemeral: false });
+    const number = interaction.options.get("number", false);
+    commandLog(interaction, "/leaderboard", "Green", [
```
Comment (Mnguyener) [inline src/commands/tutor-commands/tutor-leaderboard.ts]: Fix the name of the command to /tutor-leaderboard in the `commandLog()` function.

File: src/utils/supabase.ts [hunk 2]
```diff
@@ -1078,3 +1078,104 @@ export const getTutoringType = async (
...
+
+  if (tutorIdResponse.error) {
+    return tutorIdResponse;
+  }
+
+  const tutor_id = tutorIdResponse.data;
+  const hourResponses = await supabase.rpc("hour", { tutor_id })
+
+  if (hourResponses.error) {
+    console.log(hourResponses.error);
```
Comment (bnleft) [inline src/utils/supabase.ts]: **nitpick:** use console.error()

**Review by bnleft — APPROVED** [review]

File: src/commands/tutor-commands/tutor-leaderboard.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      {
+        name: "number",
+        value: `${number}`,
+      },
+    ]);
+
+    const errorTitle = "❌ Leaderboard Canceled!";
```
Comment (Mnguyener) [inline src/commands/tutor-commands/tutor-leaderboard.ts]: Include "Tutor" in the `errorTitle` to clarify which leader board got an error.

File: src/utils/supabase.ts (hunk 2 above)
Comment (johncoxdev) [reply src/utils/supabase.ts]: updated and pushed

File: src/commands/tutor-commands/tutor-leaderboard.ts (hunk 1 above)
Comment (johncoxdev) [reply src/commands/tutor-commands/tutor-leaderboard.ts]: updated & pushed

File: src/commands/tutor-commands/tutor-leaderboard.ts (hunk 3 above)
Comment (johncoxdev) [reply src/commands/tutor-commands/tutor-leaderboard.ts]: updated & pushed

File: src/commands/tutor-commands/tutor-leaderboard.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import { SlashCommandBuilder } from "discord.js";
+import { Command } from "../../interfaces/Command";
+import { createEmbed } from "../../utils/embeded";
+import { getTutorLeaderboard } from "../../utils/supabase";
+import { commandLog, sendError } from "../../utils/logs";
+
+const tutorleaderboardBody = (array: string[]): string => {
```
Comment (bnleft) [inline src/commands/tutor-commands/tutor-leaderboard.ts]: **suggestion:** change `array` to a more descriptive name
**comment:** cool we know `array` is an array

File: src/commands/tutor-commands/tutor-leaderboard.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  let body = "";
+  array.forEach((slot) => (body = `${body}${slot}\n`));
+  return body;
```
Comment (bnleft) [inline src/commands/tutor-commands/tutor-leaderboard.ts]: **suggestion:** refactor
**hint:** use `join()`
**comment:** KISS

**Review by bnleft — CHANGES_REQUESTED** [review]

File: src/commands/tutor-commands/tutor-leaderboard.ts (hunk 4 above)
Comment (johncoxdev) [reply src/commands/tutor-commands/tutor-leaderboard.ts]: updated & pushed

File: src/commands/tutor-commands/tutor-leaderboard.ts (hunk 5 above)
Comment (johncoxdev) [reply src/commands/tutor-commands/tutor-leaderboard.ts]: updated and pushed

File: src/commands/tutor-commands/tutor-leaderboard.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
+import { SlashCommandBuilder } from "discord.js";
+import { Command } from "../../interfaces/Command";
+import { createEmbed } from "../../utils/embeded";
+import { getTutorLeaderboard } from "../../utils/supabase";
+import { commandLog, sendError } from "../../utils/logs";
+
+const tutorleaderboardBody = (bodyContent: string[]): string => {
```
Comment (bnleft) [inline src/commands/tutor-commands/tutor-leaderboard.ts]: **suggestion:** rename function
**comment:** `tutorleaderboardBody` tells me nothing on what it does, functions do things, they typically start with verbs

**Review by bnleft — CHANGES_REQUESTED** [review]

File: src/commands/tutor-commands/tutor-leaderboard.ts (hunk 6 above)
Comment (johncoxdev) [reply src/commands/tutor-commands/tutor-leaderboard.ts]: updated & pushed

**Review by bnleft — APPROVED** [review]
nice

**Pull request merged**